apis/core/v1alpha2/featuregate_webhook.go

```diff
@@ -0,0 +1,318 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var featuregatelog = logf.Log.WithName("featuregate-resource").WithValues("apigroup", "core")
+
+var cl client.Client
+
+func getScheme() (*runtime.Scheme, error) {
+	s, err := SchemeBuilder.Build()
+	if err != nil {
+		return nil, err
+	}
+	if err := k8sscheme.AddToScheme(s); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+// Get a cached client.
+func (r *FeatureGate) getClient() (client.Client, error) {
+	if cl != nil && !reflect.ValueOf(cl).IsNil() {
+		return cl, nil
+	}
+
+	s, err := getScheme()
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	return client.New(cfg, client.Options{Scheme: s})
+}
+
+// SetupWebhookWithManager adds the webhook to the manager.
+func (r *FeatureGate) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-core-tanzu-vmware-com-v1alpha2-featuregate,mutating=false,failurePolicy=fail,groups=core.tanzu.vmware.com,resources=featuregates,versions=v1alpha2,name=vfeaturegate.kb.io
+
+var _ webhook.Validator = &FeatureGate{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
```

Comment:
For this and all the other comments, I'd like to suggest following the official Go style guide: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

> Comments documenting declarations should be full sentences...Comments should begin with the name of the thing being described and end in a period...

Reply:
resolving this based on the discussion here https://github.com/vmware-tanzu/tanzu-framework/pull/3613#discussion_r1012342860